### PR TITLE
Restructure library

### PR DESCRIPTION
**Pull request by jhuckabee**
## Status

**IN DEVELOPMENT**

## Description

This PR aims to initiate a discussion on optimizing the structure of the this package. The current approach of bundling all exports into a single file introduces challenges, especially when integrating with other libraries. Notably, this package exports common constants (e.g., Stream, Duration) that lead to naming conflicts.

The necessity for a revised structure is underscored by ongoing issues highlighted in the project's [most recent commit](https://github.com/a-wallen/google_cloudevents_dart/commit/67cfc441ae1cd7039ecebf2272b796ee79ce0b3d). That commit reflects struggles with the current export strategy, mirroring the challenges I've encountered.

The approach presented here involves segregating generated code within the lib/src directory and creating individual library files in the lib directory. Such a structure would refine import statements, enabling more granular and conflict-free integration:

```dart
import 'package:google_cloudevents_dart/events/cloud/firestore/v1.dart';

// For isolated imports where naming conflicts may occur:
import 'package:google_cloudevents_dart/protobuf.dart' show Timestamp;
```

This is loosely inspired by the structure of the [googleapis](https://github.com/google/googleapis.dart/tree/master/generated/googleapis) package.

This restructuring proposal aims not only to resolve naming conflicts but also to enhance the package's usability by allowing users to import only what they need, reducing clutter and potential conflicts.

I am eager to hear thoughts on this proposal. If this is not an ideal approach, are there alternatives that might be better suited?

Note: Instead of modifying `scripts/generate_cloudevent_library.dart`, I created a new file in `scripts/generate_cloudevent_library_files.dart`. If this proposal is something we want to move forward with, the intention is that these would get merged.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [x] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [x] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by a-wallen** [comment]
@jhuckabee I agree that this will move the package in the right direction. 

This library is closely related to `googleapis`, so it makes sense to follow the same convention. 

So, let's do it - I'm all for creating individual library files to export all the symbols without collisions even if it's a breaking change.

I think that the next steps are:
- [ ] Merge `scripts/generate_cloudevent_library.dart` and `scripts/generate_cloudevent_library_files.dart`
- [ ] Update the pubspec.yaml file with a new version `0.0.2`.
- [ ] Update CHANGELOG.md with release notes.

Is there anything that I can do to help move this contribution along?

**Comment by jhuckabee** [comment]
@a-wallen - thanks for the feedback. I've made your suggested updates. Additionally, I've added an example application showing how to use this package in a server application (using dart_frog).

File: example/routes/index.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import 'package:dart_frog/dart_frog.dart';
+import 'package:google_cloudevents_dart/events/cloud/firestore/v1.dart';
+import 'package:http/http.dart' as http;
```
Comment (a-wallen) [inline example/routes/index.dart:3]: I wonder if we should provide an example without HTTP as a dependency.

IIRC, the logic looks something like this
```dart
  final bytes = await context.request.bytes().fold(<int>[], (result, chunk) => [...result, ...chunk]);
  final event = DocumentEventData.fromBuffer(Uint8List.fromList(bytes));
```

**Review by a-wallen — APPROVED** [review]

**Comment by a-wallen** [comment]
LGTM - speaking of handling cloud events with dart_frog, check out this issue that I filed in the dart_frog repository @jhuckabee https://github.com/VeryGoodOpenSource/dart_frog/issues/1319, I'm wondering what your thoughts are.

**Pull request merged**